test-validator/src/lib.rs

```diff
@@ -108,6 +109,7 @@ pub struct TestValidatorGenesis {
     pub accountsdb_plugin_config_files: Option<Vec<PathBuf>>,
     pub accounts_db_caching_enabled: bool,
     deactivate_feature_set: HashSet<Pubkey>,
+    pub started_from_validator: bool,
```

Comment:
What the heck does `started_from_validator` mean?  This field name is super unintuitive

Reply:
I agree, and I did it ;-)
This is opposed to started from ANYTHING else. tests, local cluster tests, ledger-tool, tests of the ledger-tool, benches, ...
Testing the disk based accounts index is a thorny issue. It is created from many paths. For full confidence and coverage, we need to duplicate EVERY test env (like on the CI) that could use the accounts index to test both ways. This was not feasible. So, we allowed an env var to specify to use the disk based accounts index. BUT, there was concern that this could be read during execution of a validator. Thus, when 'starting_from_validator', we do not allow the accounts index to be affected by the env variable that allows us to fully test the disk based accounts index.
This particular test-validator tool, in @t-nelson 's opinion, is like a validator.
There is a new nightly build that will run the entire CI with the env var set, so that at least once per day we will run ALL tests with the non-default disk accounts index enabled.

Reply:
I'm open to suggestions. The goal is default to use disk accounts index in testnet asap and default for mnb in 1.10. To achieve this, we need the ability to test both code paths sufficiently and repeatedly.

Reply:
It seems like we don't need to expose this in `TestValidatorGenesis` at all, I can't imagine why a `TestValidator`-based integration test would care about this knob.  Just hard code the `started_from_validator : true` in the `AccountsDbConfig` initialization instead - that melts away most of this PR and removes a knob that probably nobody needs